internal/v2/tls_config_store/tls_config_store_test.go

```diff
@@ -217,29 +218,41 @@ func TestTLSConfigStoreServer(t *testing.T) {
 }
 
 func TestGetTLSMinMaxVersionsClient(t *testing.T) {
-	var tlsconfig s2av2pb.GetTlsConfigurationResp_ClientTlsConfiguration
-	tlsconfig.MinTlsVersion = commonpb.TLSVersion_TLS_VERSION_1_0
-	tlsconfig.MaxTlsVersion = commonpb.TLSVersion_TLS_VERSION_1_1
-	if _, _, err := getTLSMinMaxVersionsClient(&tlsconfig); err != nil {
-		t.Errorf("error in getTlsMinMaxVersionsClient logic: %v", err)
-	}
-	tlsconfig.MinTlsVersion = commonpb.TLSVersion_TLS_VERSION_1_1
-	tlsconfig.MaxTlsVersion = commonpb.TLSVersion_TLS_VERSION_1_0
-	if _, _, err := getTLSMinMaxVersionsClient(&tlsconfig); err == nil {
-		t.Errorf("getTlsMinMaxVersionsClient returned nil, should have returned error")
+	for min := commonpb.TLSVersion_TLS_VERSION_1_0; min <= commonpb.TLSVersion_TLS_VERSION_1_3; min++ {
+		for max := commonpb.TLSVersion_TLS_VERSION_1_0; max <= commonpb.TLSVersion_TLS_VERSION_1_3; max++ {
+			var tlsconfig s2av2pb.GetTlsConfigurationResp_ClientTlsConfiguration
+			tlsconfig.MinTlsVersion = min
+			tlsconfig.MaxTlsVersion = max
+			_, _, err := getTLSMinMaxVersionsClient(&tlsconfig)
+			if min <= max {
+				if err != nil {
+					t.Errorf("err = %v, expected err = nil", err)
+				}
+			} else {
+				if err == nil {
+					t.Errorf("err = nil, expected err = %v", errors.New("S2Av2 provided minVersion > maxVersion."))
+				}
+			}
+		}
```

Comment:
Let's test for the actual returned values rather than the relationship between the min and max values. Currently, the test may pass if the input values are 1.1 and 1.3 but the returned values are 1.2 and 1.3 for min and max versions. One way of doing this would be to create a mapping b/w the TLS lib versions and the proto versions and using that mapping in tests for expected vs got.